From c1c8b1bec9b28039af31b50f09e2b255efe55a68 Mon Sep 17 00:00:00 2001
From: Tiago Vignatti <tvignatti@igalia.com>
Date: Fri, 14 Oct 2022 23:34:50 +0000
Subject: [PATCH 1/3] Enable C++20 on iOS

Doing this in advance of enabling for more platforms allows monitoring
compile-time effects with less chance of adversely affecting the CQ,
while still getting the benefit of breaking a bot if someone lands a
change that breaks the build only in C++20.

Bug: 1284275
Change-Id: I0cd05cd13c94f7f597b8c39506f9a9a7f2de463a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3953137
Reviewed-by: Nico Weber <thakis@chromium.org>
Commit-Queue: Peter Kasting <pkasting@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1059534}
NOKEYCHECK=True
GitOrigin-RevId: 47957c127966524755025f10a7d49e5c0619fdec
---
 config/compiler/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 9b1e68be0..0f7cdade3 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -611,7 +611,7 @@ config("compiler") {
     # turned off we need the !is_nacl clause and the (is_nacl && is_clang)
     # clause, above.
     cflags_c += [ "-std=c11" ]
-    if (is_mac) {
+    if (is_apple) {
       # TODO(crbug.com/1284275): Switch to C++20 on all platforms.
       cflags_cc += [ "-std=c++20" ]
     } else {
-- 
2.37.1.windows.1

